pkg/network/infraconfigurators/common.go

```diff
@@ -15,7 +15,7 @@ type PodNetworkInfraConfigurator interface {
 	DiscoverPodNetworkInterface(podIfaceName string) error
 	PreparePodNetworkInterface() error
 	GenerateDomainIfaceSpec() api.Interface
-	GenerateDHCPConfig() *cache.DHCPConfig
+	GenerateNonRecoverableDHCPConfig() *cache.DHCPConfig
```

Comment:
After merging the idempotency [PR](https://github.com/kubevirt/kubevirt/pull/5943) the domain iface is not longer needed to be saved at the end of phase1 so we will be able to add to the func `WriteConfiguration() error` here that will be noop for all the bindings except bridge that will do the Domain/DHCP config generating and the saving so the interface will end up like 

```golang
type PodNetworkInfraConfigurator interface {
	DiscoverPodNetworkInterface(podIfaceName string) error
	PreparePodNetworkInterface() error
	WritePodNetworkInterfaceConfiguration() error
}
```